src/renderer/utils/utils.ts

```diff
@@ -38,18 +38,51 @@ export const useInterval = (callback: () => void, delay?: number) => {
 };
 
 // Based on https://codeforgeek.com/throttle-function-javascript/
-export const useThrottle = <Type>(
-  fn: (arg: Type) => unknown,
+export const useThrottle = <T extends (...args: any[]) => unknown>(
+  fn: T,
   delay: number
 ) => {
   const lastCalled = useRef<number>(0);
-  return (arg: Type) => {
+  return (...args: Parameters<T>) => {
     const now = new Date().getTime();
     if (now - lastCalled.current < delay) {
-      return;
+      return false;
     }
     lastCalled.current = now;
-    fn(arg);
+    fn(...args);
+    return true;
+  };
+};
+
+// This hook throttles any call to function fn. It will also rmb
+// the latest call so in the event that subsequent calls to fn stop,
+// a node interval will execute the latest call.
+export const useThrottleFuture = <T extends (...args: any[]) => any>(
```

Comment:
I find this function name to be a little confusing - possibly because I tend to associate the word "future" with asynchronous return values. This is a bit verbose but maybe something like `useThrottleAndExecuteLatest` could be more clear?